awx/ui_next/src/screens/Project/shared/ProjectForm.jsx

```diff
@@ -0,0 +1,279 @@
+        <Form autoComplete="off" onSubmit={formik.handleSubmit}>
+          <FormRow>
+            <FormField
+              id="project-name"
+              label={i18n._(t`Name`)}
+              name="name"
+              type="text"
+              validate={required(null, i18n)}
+              isRequired
+            />
+            <FormField
+              id="project-description"
+              label={i18n._(t`Description`)}
+              name="description"
+              type="text"
+            />
+            <Field
+              name="organization"
+              validate={required(
+                i18n._(t`Select a value for this field`),
+                i18n
+              )}
+              render={({ form }) => (
+                <OrganizationLookup
+                  helperTextInvalid={form.errors.organization}
+                  isValid={
+                    !form.touched.organization || !form.errors.organization
+                  }
+                  onBlur={() => form.setFieldTouched('organization')}
+                  onChange={value => {
+                    form.setFieldValue('organization', value.id);
+                    setOrganization(value);
+                  }}
+                  value={organization}
+                  required
+                />
+              )}
+            />
+            <Field
+              name="scm_type"
+              validate={required(
+                i18n._(t`Select a value for this field`),
+                i18n
+              )}
+              render={({ field, form }) => (
+                <FormGroup
+                  fieldId="project-scm-type"
+                  helperTextInvalid={form.errors.scm_type}
+                  isRequired
+                  isValid={!form.touched.scm_type || !form.errors.scm_type}
+                  label={i18n._(t`SCM Type`)}
+                >
+                  <AnsibleSelect
+                    {...field}
+                    id="scm_type"
+                    data={[
+                      {
+                        value: '',
+                        key: '',
+                        label: i18n._(t`Choose an SCM Type`),
+                        isDisabled: true,
+                      },
+                      ...scmTypeOptions.map(([value, label]) => {
+                        if (label === 'Manual') {
+                          value = 'manual';
+                        }
+                        return {
+                          label,
+                          value,
+                          key: value,
+                        };
+                      }),
+                    ]}
+                    onChange={(event, value) => {
+                      form.setFieldValue('scm_type', value);
+                      resetScmTypeFields(form);
+                    }}
+                  />
+                </FormGroup>
+              )}
+            />
+            {formik.values.scm_type !== '' && (
+              <ScmSubForm
```

Comment:
I really like that we've refactored the subforms into separate components, one for each subform type. :+1: 

That said, _this_ particular component (`ScmSubForm`) seems like an unnecessary layer of indirection. All it's doing is wrapping the collection of subforms and mapping them to the different project types. I think it would be better to inline that code here.